docsource/ops.rst

```diff
@@ -1,6 +1,236 @@
 Operations
-===========
+==============
 
-Backup and Restoration of index
-Deploying multiple indexes
+[TODO: A future PR will implement a CLI for trickle updates from a user specified time]
 
+[TODO: A future PR will include a CLI to automate step 5b by creating an index based on a search type. This would reduce potential errors. This command will also take care of step 11.]
+
+[TODO: Add CLI command for step 13 & 14 for generating events and running trickle updates once the CLI has been implemented]
+
+Reindexing Naming Conventions
+------------------------------
+
+For each type of index, "genome" for example, we may have multiple indices in ElasticSearch with the naming convention "kbase.genome_1", "kbase.genome_2" etc. where the namespace "kbase" comes from the "elastic-namespace" key-value pair in search_tools.cfg and each numeric suffix comes from the search type version (See resources/typemappings/GenomeAndAssembly.yaml.example).
+
+In order to make it easy for the client to search across all indices of a certain type, aliases are applied with the following naming convention.
+
+.. code-block:: text
+
+ alias -> [indices]
+ genome -> [genome_1, genome_2 ... genome_n]
+
+When reindexing is necessary, the index is reindexed to a new index with a new search type version. The first reindexing operation performed on genome_1 will result in a new index genome_2 that replaces genome_1 in the genome alias.
```

Comment:
E.g. 
```
gavin@login1:~$ curl ci-elasticsearch.kbase.us:9200/_cat/indices?v
health status index                                     uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   kbaseci.genome_2                          IDnemxcQRjmFUy_8OqMN5w   5   0      52982            0     91.7mb         91.7mb
green  open   kbaseci.tree_1                            5k2IY-CcSoGVAJxcF5U4tw   5   0        290            8      515kb          515kb
green  open   kbaseci.assemblycontig_1_sub              biN0f0nFQk-GEIybrnbt5A   5   0   23378937      1610718      6.8gb          6.8gb
green  open   kbaseci.media_1                           RaLTgm8FQi-Ulyb5WitBvw   5   0       2604            6      1.1mb          1.1mb
green  open   kbaseci.singleendlibrary_1                -7bVLtEZTCGGBaec1S6LMA   5   0       2686            1      1.2mb          1.2mb
green  open   kbaseci.mediacompound_1_sub               whjdSCjCSuWpJ3YCEIjbSQ   5   0      28525           85      8.3mb          8.3mb
green  open   kbaseci.genome_1                          ZC2RXqqGTCaiBMBw5XUkzw   5   0     186410          145    107.3mb        107.3mb
green  open   kbaseci.pangenomeorthologfamily_1_sub     AWpAYj4oQHuNdgYe1XBsQQ   5   0     981447        62750    297.8mb        297.8mb
green  open   kbaseci.modelreaction_1_sub               wI9TmjaJR4i2p3PYqtkcYw   5   0    1511792        13725    363.1mb        363.1mb
```

Reply:
Here's my view. There are two ways to make a change to an index. 
1. Update the spec by creating a new set of indexing rules and add a mapping with a new version number (say 2). This would result in two indexes genome_1 (existing) and genome_2 (new). 
2. Reindex genome_1 into genome_2 and then update the spec by creating a new set of indexing rules (that match up with the reindexed changes, if any) and add a mapping with a new version number (genome_2). This would also result in two indexes genome_1 (existing) and genome_2 (new). 

So in this view, both methods help move indexes from one version to the next and are therefore synonymous in this respect. Maintaining a reindex count, as I previously thought was useful, I now feel serves no value (as long as these steps are performed atomically) and only adds more steps.

Reply:
While this may work, I think it would get confusing and it would mean we would have to leave old index specs in place just to keep the ordering.  I think I would prefer something like this...

1) Make a new index spec (say genome2.yaml).  This may include multiple versions if the ws type can't be handled by a single spec (e.g. the recent update to the genome object)

2) Add this new spec as an additional index in the mapping file. (as you described)

So it may look like...
    default:
           -
               type: Genome
               version: 2
           -
               type: Genome2
               version: 2
           -
               type: GenomeFeature
               version: 2 
etc...

So then we would see something like genome_1,genome_2 and genome2_1 and genome2_2.

I'll admit this may look a little confusing but I would be worried about mixing index spec versions in the same type file.

After things are checked out, then the alias is updated to use genome2_1 and genome2_2, and, eventually, the old versions are dropped from the mapping and the old indexes can be dropped.